Base/usr/share/man/man5/Shell.md

```diff
@@ -173,6 +173,43 @@ test -f foo.txt || touch foo.txt
 rm test && echo "deleted!" || echo "failed with $?"
 ```
 
+### Immediate Functions
+A series of functions provided by the shell that can operate on AST nodes (as opposed to string arguments).
+Such functions are only available in an _immediate function_ context, i.e. inside `${...}`.
+
+Functions in this context are only allowed to be named as barewords, no escaping or string parts are allowed in such function names.
+
+The following functions are currently implemented:
+| Name          | Effect |
+| :-            | :----- |
+| `count`       | Returns the counts of its input arguments (as lists) in a list |
+| `length`      | Returns the length of its input arguments (as strings) in a list |
+| `nth`         | Returns the nth elements of its input arguments (n = first argument) |
+| `substring`   | Returns the substrings starting at arg0 with length arg1 of its inputs (as strings) |
+| `slice`       | Returns the list slice starting at arg0 with length arg1 of its inputs (as lists) |
+| `removePrefix`| Returns its input arguments, without the given prefix (arg0) - (i.e. strips the given prefix) |
+| `removeSuffix`| Similar to `removePrefix`, but strips suffixes instead |
+| `filterGlob`  | Returns the items matching the given glob (arg0) |
```

Comment:
Let's not introduce camelCase in something we have full control over.

Reply:
Luckily these are still experimental.
`removeprefix` feels dumb, `remove_prefix` looks wrong in a shell, `remove-prefix` looks ok, `RemovePrefix` is too C# to exist, `rEmOvEpReFiX` is perfection (/s)
should we just go with kebab-case?